cargo-verify/src/main.rs

```diff
@@ -0,0 +1,841 @@
+
+    let output = cmd.output()?;
+
+    let stdout = from_utf8(&output.stdout).expect("stdout is not in UTF-8");
+    let stderr = from_utf8(&output.stderr).expect("stderr is not in UTF-8");
+
+    if !output.status.success() {
+        utils::info_lines("STDERR: ", stderr.lines());
+        utils::info_lines("STDOUT: ", stdout.lines());
+        Err("`rustup show` terminated unsuccessfully")?
+    }
+
+    Ok(stdout
+        .lines()
+        .find_map(|l| l.strip_prefix("Default host:").and_then(|l| Some(l.trim())))
+        .ok_or("Unable to determine default host")?
+        .to_string())
+}
+
+// Count how many functions in fs are present in bitcode file
+fn count_symbols(opt: &Opt, bcfile: &Path, fs: &[&str]) -> usize {
+    info_at!(
+        &opt,
+        4,
+        "    Counting symbols {:?} in {}",
+        fs,
+        bcfile.to_string_lossy()
+    );
+
+    let mut cmd = Command::new("llvm-nm");
+    cmd.arg("--defined-only").arg(bcfile);
+
+    utils::info_cmd(&cmd, "llvm-nm");
+
+    let output = cmd.output().expect("Failed to execute `llvm-nm`");
+
+    let stdout = from_utf8(&output.stdout).expect("stdout is not in UTF-8");
+    // let stderr = from_utf8(&output.stderr).expect("stderr is not in UTF-8");
+
+    // TODO:
+    // if ! output.status.success() {
+
+    let count = stdout
+        .lines()
+        .map(|l| l.split(" ").collect::<Vec<_>>())
+        .filter(|l| l.len() == 3 && l[1] == "T" && fs.iter().any(|f| f == &l[2]))
+        .count();
+
+    info_at!(&opt, 4, "    Found {} functions", count);
+    count
+}
+
+// Generate a list of tests in the crate by parsing the output of `cargo test --
+// --list`
+fn list_tests(opt: &Opt, target: &str) -> CVResult<Vec<String>> {
+    let mut cmd = Command::new("cargo");
+    cmd.arg("test").arg("--manifest-path").arg(&opt.cargo_toml);
+
+    if !opt.features.is_empty() {
+        cmd.arg("--features").arg(opt.features.join(","));
+    }
+
+    cmd.arg(format!("--target={}", target))
+        .args(vec!["-v"; opt.verbosity])
+        .envs(get_build_envs(&opt)?)
+        .args(&["--", "--list"]);
+    // .arg("--exclude-should-panic")
+    // .env("PATH", ...)
+
+    utils::info_cmd(&cmd, "cargo");
+
+    let output = cmd.output()?;
+
+    let stdout = from_utf8(&output.stdout).expect("stdout is not in UTF-8");
+    let stderr = from_utf8(&output.stderr).expect("stderr is not in UTF-8");
+
+    if false && !output.status.success() {
+        utils::info_lines("STDOUT: ", stdout.lines());
+        utils::info_lines("STDERR: ", stderr.lines());
+        Err("Couldn't get list of tests")?;
+    }
+
+    lazy_static! {
+        static ref TEST: Regex = Regex::new(r"(\S+):\s+test\s*$").unwrap();
+    }
+
+    let tests = stdout
+        .lines()
+        .filter_map(|l| {
+            TEST.captures(l)
+                .map(|caps| caps.get(1).unwrap().as_str().into())
+        })
+        .collect();
+
+    Ok(tests)
+}
+
+// Find a function defined in LLVM bitcode file
```

Comment:
The comment for the function with the same name in Python script talks about what happens with the hash at the end of the mangled name.

More broadly, I think it would be best to copy the comments from the Python script exactly as they are and then improve them instead of deleting them or writing something else.

Reply:
Sorry about the comments. I mostly tried to do just that, at least in my head I thought I was doing that :).